### DISCUSSION TITLE
Exporting a submission as a PDF

### DISCUSSION THREAD

**Discussion by stuh** [verbb/formie#1273] [Q&A] [answered]
Currently evaluating Formie for a client as part of a recruitment system. The only issue we can see is their requirement to be able to export a submission as a PDF from the admin panel. Im thinking of some approaches to solving this and would love your thoughts or suggestion on how best to do it.

1) create our own admin only template which will list all submissions. The link for each opens a twig template which loads the form and we use a PDF plugin to render the page as a PDF. One issue with this is that the form is multi page so we’d need to override this admin templates CSS to display it as one page. Or can we override the Form Settings from the template to turn off multi page for this template?

2) same as above but we use a custom styled summary field to list all the data in one shot and hide the rest.

3) we somehow are able to give URL access to the email notification PDF template.

Thanks, another great Verbb plug-in!

**Comment by engram-design** [comment, marked as answer]
My other suggestion would be to create a custom exporter class (if you've got some module / PHP know-how), which when registered, you could add your own logic for how an export is run.

```php
use verbb\formie\elements\Submission;
use craft\events\RegisterElementExportersEvent;
use yii\base\Event;

Event::on(
    Submission::class,
    Submission::EVENT_REGISTER_EXPORTERS,
    function(RegisterElementExportersEvent $event) {
        $event->exporters[] = MyElementExporter::class;
    }
);
```

More info https://craftcms.com/docs/3.x/extend/element-exporter-types.html

**Reply by stuh** [reply]
Thanks for the quick response. This is great, I didn’t know we could have custom exporter types. Thank you.

**Reply by jubalj** [reply]
Hi Stuh,

Any chance your solution will be public domain or usable by others? I think I have a similar requirement and had put in a feature request ( https://github.com/verbb/formie/issues/1213). 

Thanks
Jubal

**Reply by stuh** [reply]
Hi Jubal, I've not began work on it yet I'm afraid but when I do I'll post back here with what we come up with.

Should be within the next week.

Cheers

**Reply by jubalj** [reply]
Hi stuh,

I wanted to follow up to see if you had a chance to get pdf exports working and if you would be happy posting your code please!

Warm wishes
Jubal

**Comment by stuh** [comment]
Hi Jubal, we actually went with a different direction and ended up creating some templates which listed the submissions and used https://github.com/enupal/snapshot to generate PDF's for them. Functionally very similar to your feature request #1213 

Hope this helps.

**Comment by engram-design** [comment]
Added in [4.0.0-beta.8](https://github.com/verbb/formie/releases/tag/4.0.0-beta.8).

**Comment by jubalj** [comment]
amazing work! thanks very much.
